bindings/python/build_tools/cmake/iree_pybind_cc_library.cmake

```diff
@@ -0,0 +1,123 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+include(CMakeParseArguments)
+
+# iree_pybind_cc_library()
+#
+# CMake function to imitate Bazel's pybind_cc_library rule.
+#
+# Parameters:
+# NAME: name of target
+# HDRS: List of public header files for the library
+# SRCS: List of source files for the library
+# DEPS: List of other libraries to be linked in to the binary targets
+# COPTS: List of private compile options
+# DEFINES: List of public defines
+# INCLUDES: Include directories to add to dependencies
+# LINKOPTS: List of link options
+# TYPE: Type of library to be crated: either "MODULE", "SHARED" or "STATIC" (default).
+# TESTONLY: When added, this target will only be built if user passes -DIREE_BUILD_TESTS=ON to CMake.
+
+function(iree_pybind_cc_library)
+
+  cmake_parse_arguments(
+    _RULE
+    "TESTONLY"
+    "NAME"
+    "HDRS;SRCS;COPTS;DEFINES;LINKOPTS;DEPS;INCLUDES;TYPE"
+    ${ARGN}
+  )
+
+  if(NOT _RULE_TESTONLY OR IREE_BUILD_TESTS)
+    # Prefix the library with the package name, so we get: iree_package_name.
+    iree_package_name(_PACKAGE_NAME)
+    set(_NAME "${_PACKAGE_NAME}_${_RULE_NAME}")
+
+    if(NOT _RULE_TYPE)
+      set(_RULE_TYPE "STATIC")
+    endif()
+
+    string(TOUPPER "${_RULE_TYPE}" _uppercase_RULE_TYPE)
+    if(NOT _uppercase_RULE_TYPE MATCHES "^(STATIC|SHARED|MODULE)")
+      message(FATAL_ERROR "Unsported library TYPE for iree_pybind_cc_library: ${_RULE_TYPE}")
+    endif()
+
+    add_library(${_NAME} ${_uppercase_RULE_TYPE} "")
+    target_sources(${_NAME}
+      PRIVATE
+        ${_RULE_SRCS}
+        ${_RULE_HDRS}
+    )
+    target_include_directories(${_NAME}
+      PUBLIC
+        "$<BUILD_INTERFACE:${IREE_COMMON_INCLUDE_DIRS}>"
+        "$<BUILD_INTERFACE:${_RULE_INCLUDES}>"
+      PRIVATE
+        ${PYBIND11_INCLUDE_DIR}
+        ${pybind11_INCLUDE_DIR} 
```

Comment:
Do we really need both uppercase and lowercase `pybind11_INCLUDE_DIR`? Seems weird, though I did find https://github.com/pybind/pybind11/blob/07e225932235ccb0db5271b0874d00f086f28423/tools/pybind11Tools.cmake#L138-L140

Reply:
I am not sure, more or less just have taken this over. That is one of the things I wanted to test when the bindings get fully available via CMake.

Reply:
Added a TODO on that.